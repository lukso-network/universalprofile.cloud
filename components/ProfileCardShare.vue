<script setup lang="ts">
type Props = {
  profile?: Profile
}

defineProps<Props>()
</script>

<template>
  <div class="absolute right-4 top-4 flex flex-col items-end gap-2">
    <ProfileCardShareLink :profile="profile" />
    <ProfileCardShareUniversalPage :profile="profile" />
    <ProfileCardShareUniversalSwaps :profile="profile" />
  </div>
</template>
