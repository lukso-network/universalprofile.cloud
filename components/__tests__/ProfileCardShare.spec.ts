import { renderSuspended } from '@nuxt/test-utils/runtime'
import { describe, expect, it } from 'vitest'

import ProfileCardShare from '@/components/ProfileCardShare.vue'

describe('ProfileCardShare', () => {
  const profile: Profile = {
    address: '0x1234567890abcdef1234567890abcdef12345678',
    profileLink: {
      resolved:
        'https://universalswaps.io/social/0x1234567890abcdef1234567890abcdef12345678',
      link: 'https://universalswaps.io/social/{address}',
      address: '0x1234567890abcdef1234567890abcdef12345678',
      checksummed: '0x1234567890abcdef1234567890abcdef12345678',
      isResolved: true,
    },
  }

  it('should render', async () => {
    const component = await renderSuspended(ProfileCardShare, {
      props: { profile },
    })

    expect(component.html()).toMatchSnapshot()
  })
})
