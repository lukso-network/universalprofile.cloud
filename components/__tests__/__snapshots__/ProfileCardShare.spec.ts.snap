// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProfileCardShare > should render 1`] = `
"<div id="test-wrapper">
  <div class="absolute right-4 top-4 flex flex-col items-end gap-2">
    <div>
      <lukso-button size="small" variant="secondary" custom-class="px-0 overflow-hidden size-[30px]">
        <lukso-icon size="small" name="transaction-send" class="mx-1.5"></lukso-icon>
        <div class="paragraph-inter-12-medium transition-width duration-500 ease-out" style="width: auto;">
          <div class="inline-flex items-center text-nowrap">
            <div class="m-1 mr-1.5 text-neutral-50">universalswaps.io/social/{address}</div>
          </div>
          <div class="inline-flex">
            <div class="mr-1.5"></div>
          </div>
        </div>
      </lukso-button>
    </div>
    <lukso-tooltip variant="white" text="universal.page" placement="left">
      <lukso-button size="small" variant="secondary" custom-class="px-0 overflow-hidden size-full" class="group size-[30px]"><img src="/images/universalpage-logo.png" alt="" class="h-4 grayscale transition group-hover:grayscale-0"></lukso-button>
    </lukso-tooltip>
    <lukso-tooltip variant="white" text="universalswaps.io" placement="left">
      <lukso-button size="small" variant="secondary" custom-class="px-0 overflow-hidden size-full" class="group size-[30px]"><img src="/images/universalswaps-logo.png" alt="" class="max-h-4 max-w-[20px] grayscale transition group-hover:grayscale-0"></lukso-button>
    </lukso-tooltip>
  </div>
</div>"
`;
